### PR TITLE
Add Compose code for Analytics

### PR DESCRIPTION
**Pull request by lehcar09**
Changes made:

Add Jetpack Compose option in EntryChoiceActivity
Removed business logic from MainActivity
Added FirebaseAnalyticsViewModel for business logic
Added ComposeMainActivity for the Jetpack Compose UI

### CODE REVIEW

File: analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/MainActivity.kt [hunk 1]
```diff
@@ -167,10 +132,7 @@ class MainActivity : AppCompatActivity() {
             startActivity(sendIntent)
 
             // [START custom_event]
-            firebaseAnalytics.logEvent("share_image") {
-                param("image_name", name)
-                param("full_text", text)
-            }
+            viewModel.recordShare(name, text)
```
Comment (thatfiredev) [inline analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/MainActivity.kt:135]: nit: can we rename this to `recordShareEvent` ?

File: analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/FirebaseAnalyticsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        firebaseAnalytics.logEvent("share_image") {
+            param("image_name", imageTitle)
+            param("full_text", text)
+        }
+    }
+
+    fun recordScreenView(screenName: String) {
+        firebaseAnalytics.logEvent(FirebaseAnalytics.Event.SCREEN_VIEW) {
+            param(FirebaseAnalytics.Param.SCREEN_NAME, screenName)
+            param(FirebaseAnalytics.Param.SCREEN_CLASS, "MainActivity")
```
Comment (thatfiredev) [inline analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/FirebaseAnalyticsViewModel.kt:57]: I think we should also pass the screen class as a parameter to this function, as it makes it easier to reuse it from a different screen.

File: analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/MainActivity.kt (hunk 1 above)
Comment (lehcar09) [reply analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/MainActivity.kt:135]: Yes. Will rename API.

File: analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/FirebaseAnalyticsViewModel.kt (hunk 2 above)
Comment (lehcar09) [reply analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/FirebaseAnalyticsViewModel.kt:57]: Adding the screen class parameter in the API

**Comment by lehcar09** [comment]
CLA is failing due to incorrect user used on recent commit. Closing this and creating new one with the fix from review.

**Pull request closed without merging**